### ISSUE TITLE
Vulkan: SSAO debug draw mode freezes after disabling SSAO

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#39875]
**Godot version:** Git 175d43738ac58afcf2d0531177eefeb260c4f296

**OS/device including version:** Fedora 31, GeForce GTX 1080 (NVIDIA 440.82)

**Issue description:**

After enabling SSAO and switching to the SSAO debug draw mode in the 3D viewport, disabling SSAO in WorldEnvironment will cause the viewport to freeze until the user switches back to another draw mode.

The draw mode should automtaically be changed to "Draw Normal" once SSAO is disabled. Also, when SSAO is disabled, it shouldn't be possible to switch to it in the first place (it should be grayed out with a tooltip).

**Steps to reproduce:**

- Add a WorldEnvironment node and create a new Environment.
- Enable SSAO in the Environment (already done in the MRP below).
- Click the **Perspective** button in the top-left corner and change the draw mode to **SSAO** in the advanced draw modes list.
  - *Note:* On Linux, use arrow keys to access the second dropdown so it doesn't close when your mouse hovers it.
- Disable SSAO in the Environment. Try to move the camera, see that nothing changes.

**Minimal reproduction project:** [test_ssao_draw_mode_bug.zip](https://github.com/godotengine/godot/files/4840367/test_ssao_draw_mode_bug.zip)

**Comment by clayjohn** [comment]
I think this is more of an enhancement actually. If you turn off SSAO, it stops updating, so it is as of your screen has frozen in place. 

Changing viewport modes based on environment settings is an enhancement IMO.

**Comment by BastiaanOlij** [comment]
Indeed, the debug check simply checks whether there is an SSAO buffer, which will be created when SSAO is turned on, but is not cleaned up if it is turned off.

I've left this alone even with moving everything to our render buffers implementation but I was thinking about this. It would be relatively easy to add a cleanup step if SSAO is turned off (and same with SSIL) and that would void the debug setting.

**Comment by clayjohn** [comment]
> Indeed, the debug check simply checks whether there is an SSAO buffer, which will be created when SSAO is turned on, but is not cleaned up if it is turned off.
> 
> I've left this alone even with moving everything to our render buffers implementation but I was thinking about this. It would be relatively easy to add a cleanup step if SSAO is turned off (and same with SSIL) and that would void the debug setting.

I think that would be a good idea. We probably shouldn't be holding on to resources if effects are turned off.

**Comment by KoBeWi** [comment]
Still valid in 4.4
